rtmp_to_hls/01_Introduction.md

```diff
@@ -0,0 +1,32 @@
+We are glad you have decided to continue your journey into the oceans of multimedia with us!
+In this tutorial you will be able to see the Membrane in action - as we will prepare a solution responsible for converting
+the incoming RTMP stream into an HLS stream, ready to be easily served on the internet.
+
+## What do we want to achieve?
+
+Imagine that we need to build a video streaming platform. That means - we want a particular system user, the streamer, to stream its multimedia on a server, where it will be available for other participants - the viewers.
+We need to take advantage of the fact, that there will be only one person streaming, and take into consideration, that there might be multiple, possibly many, viewers. What's the best solution for such a use case?
+Well, as the tutorial name suggests - we can use RTMP for the streamer to stream its multimedia to the server, and make them accessible for the viewers by broadcasting them via HLS protocol!
+Easier said than done - how to achieve that? Well, with the Membrane Framework this will be easier than you can expect!
+As a final product of this tutorial, we want to have a completely working solution for broadcasting a stream. It will consist of two parts:
+
+- The pipeline, responsible for receiving an RTMP stream and preparing an HLS stream out of it
+- The web player, capable of playing the HLS stream
+
+The Membrane will find its ground to show off in the first part we will prepare a Membrane's Pipeline for converting the stream.
+When it comes to the web player, we will use an existing solution - the [HLS.js](https://github.com/video-dev/hls.js/) player.
+
+## Why one would need such a solution?
+
+You might wonder why one would need to convert an RTMP stream into an HLS stream - couldn't we simply make the streamer broadcast its multimedia with the RTMP to all the viewers?
+Technically speaking we could...but surprisingly it wouldn't be the easiest solution, since each of the viewers would need to act as an RTMP server. And definitely, it wouldn't be a solution that would scale - since RTMP is based on TCP, there is no way to broadcast the stream, and therefore the streamer would need to perform a three-way handshake with each of the viewers.
```

Comment:
Yeah, you are right - one of the problems is that we need to keep the connection alive. However, I don't consider it to be the most problematic issue.
What I've meant was that it is the streamer, who needs to connect with each of the viewers directly (and that it was I described as "performing a three-way handshake with each of the viewers").
Such a solution doesn't scale well, since the streamer might not have enough resources to start and mantain the connection with thousands of the viewers, in constrast to the HTTP server. In addition, HTTP - based communication might be less resource-consuming when compared with RTMP, as you have mentioned. 

I agree that that paragraph needs to be rewritten.

Reply:
done